src/main/java/org/jabref/preferences/SearchPreferences.java

```diff
@@ -1,103 +1,63 @@
-
-    private static final String SEARCH_DIALOG_HEIGHT = "searchDialogHeight";
-    private static final String SEARCH_DIALOG_WIDTH = "searchDialogWidth";
-    private static final String SEARCH_DIALOG_POS_X = "searchDialogPosX";
-    private static final String SEARCH_DIALOG_POS_Y = "searchDialogPosY";
+    private final SearchDisplayMode searchDisplayMode;
+    private final boolean isCaseSensitive;
+    private final boolean isRegularExpression;
 
-    private final JabRefPreferences preferences;
-
-    public SearchPreferences(JabRefPreferences preferences) {
-        this.preferences = Objects.requireNonNull(preferences);
+    public SearchPreferences(SearchDisplayMode searchDisplayMode, boolean isCaseSensitive, boolean isRegularExpression) {
+        this.searchDisplayMode = searchDisplayMode;
+        this.isCaseSensitive = isCaseSensitive;
+        this.isRegularExpression = isRegularExpression;
     }
 
-    public static void putDefaults(Map<String, Object> defaults) {
-        defaults.put(SEARCH_DISPLAY_MODE, SearchDisplayMode.FILTER.toString());
-        defaults.put(SEARCH_CASE_SENSITIVE, Boolean.FALSE);
-        defaults.put(SEARCH_REG_EXP, Boolean.FALSE);
-
-        defaults.put(SEARCH_DIALOG_WIDTH, 650);
-        defaults.put(SEARCH_DIALOG_HEIGHT, 500);
-        defaults.put(SEARCH_DIALOG_POS_X, 0);
-        defaults.put(SEARCH_DIALOG_POS_Y, 0);
-    }
-
-    public SearchDisplayMode getSearchMode() {
-        try {
-            return SearchDisplayMode.valueOf(preferences.get(SEARCH_DISPLAY_MODE));
-        } catch (IllegalArgumentException ex) {
-            // Should only occur when the searchmode is set directly via preferences.put and the enum was not used
-            return SearchDisplayMode.valueOf((String) preferences.defaults.get(SEARCH_DISPLAY_MODE));
-        }
-    }
-
-    public SearchPreferences setSearchMode(SearchDisplayMode searchDisplayMode) {
-        preferences.put(SEARCH_DISPLAY_MODE, Objects.requireNonNull(searchDisplayMode).toString());
-        return this;
+    public SearchDisplayMode getSearchDisplayMode() {
+        return searchDisplayMode;
     }
 
     public boolean isCaseSensitive() {
-        return preferences.getBoolean(SEARCH_CASE_SENSITIVE);
-    }
-
-    public SearchPreferences setCaseSensitive(boolean isCaseSensitive) {
-        preferences.putBoolean(SEARCH_CASE_SENSITIVE, isCaseSensitive);
-        return this;
+        return isCaseSensitive;
     }
 
     public boolean isRegularExpression() {
-        return preferences.getBoolean(SEARCH_REG_EXP);
+        return isRegularExpression;
     }
 
-    public SearchPreferences setRegularExpression(boolean isRegularExpression) {
-        preferences.putBoolean(SEARCH_REG_EXP, isRegularExpression);
-        return this;
+    public Builder getBuilder() {
+        return new Builder(this);
     }
 
-    public int getSeachDialogWidth() {
-        return preferences.getInt(SEARCH_DIALOG_WIDTH);
-    }
+    public static class Builder {
```

Comment:
Is the builder really necessary? I think the `with*` methods can be added directly to the main class without any problems.

Reply:
I'll change it, I originally just wanted to stick close to the builder pattern, as I don't have much experience on it, so this was somewhat a study project for me :wink: